### PR TITLE
Add basic support for isolated projects to 1.x

### PR DESCRIPTION
**Pull request by technoir42**
Since 2.0 seems quite far away, it'd be great to add support for isolated projects to 1.x branch. It's quite simple to do and works well on a couple of my projects.

### CODE REVIEW

File: detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt [hunk 1]
```diff
@@ -25,8 +25,7 @@ class DetektPlugin : Plugin<Project> {
 
         extension.reportsDir = project.extensions.getByType(ReportingExtension::class.java).file("detekt")
 
-        val defaultConfigFile =
-            project.file("${project.rootProject.layout.projectDirectory.dir(CONFIG_DIR_NAME)}/$CONFIG_FILE")
+        val defaultConfigFile = project.rootDir.resolve("$CONFIG_DIR_NAME/$CONFIG_FILE")
```
Comment (technoir42) [inline detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt:28]: Another, more modern alternative is `project.isolated.rootProject.projectDirectory.file()` which requires Gradle 8.8+, but `rootDir` seems to work fine for this purpose.

**Comment by cortinico** [comment]
> Since 2.0 seems quite far away, it'd be great to add support for isolated projects to 1.x branch. It's quite simple to do and works well on a couple of my projects.

Is this a pick of another PR @technoir42 ?

**Comment by technoir42** [comment]
@cortinico nope, made from scratch

**Comment by cortinico** [comment]
> @cortinico nope, made from scratch

Ok can we have this PR against main then?

**Comment by technoir42** [comment]
@cortinico main branch does seem to support isolated projects already, but I can the tests for isolated projects in another PR targeted at main.

**Comment by 3flex** [comment]
I don't have time for a close review for a couple of days, but does this fix anything that wasn't already fixed by https://github.com/detekt/detekt/pull/7394?

**Comment by 3flex** [comment]
@technoir42:

> I can add the tests for IP in another PR targeted at main

This is my preference. Eventually we should enable IP for every Gradle functional test but this isn't possible until the Kotlin Gradle plugin adds multiplatform support (hopefully in Kotlin 2.1.0, see child tasks of [KT-57279](https://youtrack.jetbrains.com/issue/KT-57279/Support-Gradle-Project-Isolation-Feature-for-Kotlin-Multiplatform))

And I think we should also close this PR. I don't think IP support is important enough to make this change in the 1.23.x line, especially because the support will be quite incomplete (see other limitations listed in https://github.com/detekt/detekt/pull/7394).

**Comment by technoir42** [comment]
@3flex I added tests for IP in #7535.

> And I think we should also close this PR. I don't think IP support is important enough to make this change in the 1.23.x line, especially because the support will be quite incomplete (see other limitations listed in https://github.com/detekt/detekt/pull/7394).

The cost of backporting changes seems to be trivial and at least some users will benefit from it. For us, detekt is the last plugin which blocks enabling IP and has no workaround due to the internal use of `rootProject.layout`. So I'd appreciate it if a 1.23.x release could be made with basic IP support, even multiplatform and report merging wouldn't be supported.

**Comment by 3flex** [comment]
@cortinico it's your call whether you want to merge this given you've been maintaining this branch. I don't plan to review in detail though.

**Comment by technoir42** [comment]
I updated the PR to be based on cherry-picks from #7340, #7394, #7483 and #7535.

@BraisGabin @cortinico would it be possible to review this and cut a release from 1.23.x line with these changes?
detekt is the last blocker for us as early adopters of IP.

**Review by cortinico — APPROVED** [review]
Code LGTM 👍 Let's merge it and I should be able to make a 1.x release in 1/2 weeks (sorry for the delay)

**Comment by technoir42** [comment]
Hey @cortinico, just wanted to kindly ping you about this

> I should be able to make a 1.x release in 1/2 weeks (sorry for the delay)

**Comment by cortinico** [comment]
Yes I'm spent the day yesterday in trying to release a new version from the 1.x branch. The problem is that every pick I do has merge conflicts that need to be resolved manually. This is going to take me a bit longer than expected.

**Pull request merged**